### PR TITLE
eqwidth + convertADU

### PR DESCRIPTION
**Pull request by VBremaud**

### CODE REVIEW

File: spectractor/extractor/spectrum.py [hunk 1]
```diff
@@ -851,9 +854,40 @@ def detect_lines(lines, lambdas, spec, spec_err=None, fwhm_func=None, snr_minlev
             # save fit results
             line.fitted = True
             line.fit_lambdas = lambdas[index]
+            
+            x_norm = rescale_x_for_legendre(lambdas[index])
+
+            X=np.arange(int((peak_pos - 5*np.abs(popt[bgd_npar + 3 * j + 2]))*10)/10,int((peak_pos + 5*np.abs(popt[bgd_npar + 3 * j + 2]))*10)/10,0.1)
```
Comment (jeremyneveu) [inline spectractor/extractor/spectrum.py:860]: remove int() and *10/10

File: spectractor/extractor/spectrum.py [hunk 2]
```diff
@@ -851,9 +854,40 @@ def detect_lines(lines, lambdas, spec, spec_err=None, fwhm_func=None, snr_minlev
...
+                X_norm = X_norm / np.max(lambdas[index]-middle)
+
+            test=0
+            for loop in range(len(lambdas)):
+                if lambdas[loop]>int((peak_pos - 5*np.abs(popt[bgd_npar + 3 * j + 2]))*10)/10 and test==0:
+                    jmin=loop-1
+                    test=1
+                if lambdas[loop]>int((peak_pos + 5*np.abs(popt[bgd_npar + 3 * j + 2]))*10)/10:
+                    jmax=loop+1
+                    break
```
Comment (jeremyneveu) [inline spectractor/extractor/spectrum.py:874]: jmin = np.argmin(np.abs(lambdas - (peak_pos - 5*np.abs(popt[bgd_npar + 3 * j + 2]))))

File: spectractor/extractor/spectrum.py [hunk 3]
```diff
@@ -851,9 +854,40 @@ def detect_lines(lines, lambdas, spec, spec_err=None, fwhm_func=None, snr_minlev
             # save fit results
             line.fitted = True
             line.fit_lambdas = lambdas[index]
+            
+            x_norm = rescale_x_for_legendre(lambdas[index])
+
+            X=np.arange(int((peak_pos - 5*np.abs(popt[bgd_npar + 3 * j + 2]))*10)/10,int((peak_pos + 5*np.abs(popt[bgd_npar + 3 * j + 2]))*10)/10,0.1)
+
+            middle = 0.5 * (np.max(lambdas[index]) + np.min(lambdas[index]))
+            X_norm = X - middle
```
Comment (jeremyneveu) [inline spectractor/extractor/spectrum.py:863]: rename X into x_int and X_norm into x_int_norm

File: spectractor/extractor/spectrum.py [hunk 4]
```diff
@@ -851,9 +854,40 @@ def detect_lines(lines, lambdas, spec, spec_err=None, fwhm_func=None, snr_minlev
...
+            for loop in range(len(lambdas)):
+                if lambdas[loop]>int((peak_pos - 5*np.abs(popt[bgd_npar + 3 * j + 2]))*10)/10 and test==0:
+                    jmin=loop-1
+                    test=1
+                if lambdas[loop]>int((peak_pos + 5*np.abs(popt[bgd_npar + 3 * j + 2]))*10)/10:
+                    jmax=loop+1
+                    break
+
+            spectr=interp1d(lambdas[jmin:jmax],spec[jmin:jmax])
+            spectr_data=spectr(X)
```
Comment (jeremyneveu) [inline spectractor/extractor/spectrum.py:877]: merge in one line ?

File: spectractor/parameters.py [hunk 5]
```diff
@@ -121,7 +121,7 @@ def __getattr__(name):
 TIME_UNIT = 1 * units.s  # flux for 1 second
 hc = const.h * const.c  # h.c product of fontamental constants c and h
 wl_dwl_unit = units.nanometer ** 2  # lambda.dlambda  in wavelength in nm
-g_disperser_ronchi = 0.2  # theoretical gain for order+1 : 10%
+g_disperser_ronchi = 1  # theoretical gain for order+1 : 10%
```
Comment (jeremyneveu) [inline spectractor/parameters.py:124]: remove g_diserpser_ronchi

File: spectractor/extractor/spectroscopy.py [hunk 6]
```diff
@@ -2,6 +2,7 @@
 from scipy.interpolate import interp1d
 import numpy as np
 import copy
+import csv
```
Comment (jeremyneveu) [inline spectractor/extractor/spectroscopy.py:5]: remove csv file

File: spectractor/extractor/spectroscopy.py [hunk 7]
```diff
@@ -331,6 +334,11 @@ def plot_detected_lines(self, ax=None, print_table=False):
         lambdas = np.zeros(1)
         rows = []
         j = 0
+
+        fichier=open("detedted_lines.csv","wt")
+        Detected_lines=csv.writer(fichier,delimiter=";")
+        Detected_lines.writerow(["Line", "Tabulated", "Detected", "Shift", "FWHM", "Amplitude", "SNR", "Chisq","Eqwidth_mod","Eqwidth_data"])
+
```
Comment (jeremyneveu) [inline spectractor/extractor/spectroscopy.py:341]: use astropy.Table to write a csv file

File: spectractor/extractor/spectroscopy.py [hunk 8]
```diff
@@ -283,7 +283,7 @@ def plot_atomic_lines(self, ax, color_atomic='g', color_atmospheric='b', fontsiz
                             xycoords='axes fraction', color=color, fontsize=fontsize)
         return ax
 
-    def plot_detected_lines(self, ax=None, print_table=False):
+    def plot_detected_lines(self,output_file_name,overwrite=False,ax=None, print_table=False):
```
Comment (VBremaud) [inline spectractor/extractor/spectroscopy.py:286]: Problem with arguments for other uses of the function.
 L209 , L870, spectrum.py

File: spectractor/parameters.py (hunk 5 above)
Comment (jeremyneveu) [reply spectractor/parameters.py]: also removed from config.py

**Pull request merged**